cwltool/expression.py

```diff
@@ -53,14 +53,28 @@ def scanner(scan):  # type: (Text) -> List[int]
     SINGLE_QUOTE = 4
     DOUBLE_QUOTE = 5
     BACKSLASH = 6
+    COMMENT = 7
 
     i = 0
     stack = [DEFAULT]
     start = 0
+    com_check = ''
     while i < len(scan):
         state = stack[-1]
         c = scan[i]
 
+        if state == COMMENT:
+            if c == '\n':
+                stack.pop()
+                com_check = ''
+            pass
+        com_check += c
+        if com_check == '//':
+            stack.append(COMMENT)
+            pass
+        elif com_check != '/':
+            com_check = ''
+
         if state == DEFAULT:
```

Comment:
FYI: This might allow `//` style comments inside CWL paramenter references, which is contrary to the CWL v1.0.x standards.

Reply:
Ah I see didn't realize this was used to parse `valueFrom`'s thought it was just for ExpressionTools. Perhaps I should close this and return when I have a fuller picture of the parsing environment

Reply:
This method is used to extract the text inside `${}` but there is a separate regex that determines if the text is a plain parameter reference or arbitrary javascript.  So I think adding comment parsing is okay.

Reply:
(The purpose is to parse just enough javascript grammar to be able to recognize where the expression begins and ends)

Reply:
Looks like an issue here:
```
Test failed: /home/jenkins/workspace/cwltool-pr-conformance-multiver/version/v1.0/cwltool-venv2/bin/cwltool --outdir=/tmp/tmppDHNEh --quiet v1.0/test-cwl-out2.cwl v1.0/empty.json
Test support for reading cwl.output.json when running in a Docker container
and just 'location' is provided.
```
Perhaps you point me in the direction of this failing test?

Reply:
Looks like you triggered on the `//` in a `file://` URI:

```
Workflow error, try again with --debug for more information:
:1:1: Expression evaluation error:
      Substitution error, unfinished block starting at position 0: echo foo > foo && echo '{"foo":
      {"location": "file://$(runtime.outdir)/foo", "class": "File"} }' > cwl.output.json
```

Reply:
If that's happening, the test for `//` is happening in the wrong parser state.  I'll dig into it when I get a chance.

Reply:
@tetron Can you take a look? Having a full URI in the `format` field is leading to `Dockstore.cwl:45:5: Substitution error, unfinished block starting at position 0: 'http://edamontology.org/data_3671' stack was [0, 7]`

https://github.com/common-workflow-language/cwltool/pull/739/checks?check_run_id=3461661076#step:9:352